workshops/models.py

```diff
@@ -296,17 +296,20 @@ def get_by_ident(ident):
         YYYY-MM-DD-site slug, and indicates an event whose dates are firm.
         The real indicator is the 'published' flag.
         '''
-
-        # match event ID
-        if re.match(r'^\d+$', ident):
-            return Event.objects.get(id=ident)
-
-        # match event slug in format YYYY-MM-**** (ie. this works for
-        # YYYY-MM-DD-**** too)
-        if re.match(r'^\d{4}-\d{2}-.+$', ident):
-            return Event.objects.get(slug=ident)
-
-        raise ObjectDoesNotExist(ident)
+        kwargs = {}
+        try:
+            # first try to match using ID
+            kwargs["id"] = int(ident)
+        except ValueError:
+            # match event slug in format YYYY-MM-****
+            #(ie. this works for YYYY-MM-DD-**** too)
+            if re.match(r'^\d{4}-\d{2}-.+$', ident):
+                kwargs["slug"] = ident
+            else:
+                raise ObjectDoesNotExist(ident)
+
+        # may throw DoesNotExist exception
+        return Event.objects.get(**kwargs)
```

Comment:
Hmmm, I have mixed feelings about this method.

I don't like that this method may throw two different exceptions (but it's not fault of your code but rather of the original code). But I like the that we have (partial) duck typing here.

What I think we should do here is something like:
```python
def get_by_ident(ident):
    try:
        return Event.objects.get(pk=int(ident))
    except ValueError:
        return Event.objects.get(slug=ident)
```

It's simpler, it's more Pythonic (I hope), it doesn't include slug validation where that validation should not happen.

If we really need to validate the slug format, we should do that in `save()`.

@wking, any thoughts?